src/neighborsearch/neighborhood.jl

```diff
@@ -26,9 +26,16 @@ function NeighborhoodSearch(object::O, neigh::N) where {O,N}
   else
     nothing
   end
+
   NeighborhoodSearch{O,N,typeof(tree)}(object, neigh, tree)
 end
 
+searchinds(xₒ::AbstractVector, method::NeighborhoodSearch{O,N,T}) where {O,N<:BallNeighborhood,T} =
+  inrange(method.tree, xₒ, radius(method.neigh))
+
+searchinds(xₒ::AbstractVector, method::NeighborhoodSearch{O,N,T}) where {O,N<:EllipsoidNeighborhood,T} =
+  inrange(method.tree, xₒ, one(eltype(xₒ)))
+
 # search method for any neighborhood
 function search(xₒ::AbstractVector, method::NeighborhoodSearch; mask=nothing)
```

Comment:
Perhaps we can also omit the type annotation of the first argument here too. Could you please double check?

Reply:
Omitting this one leads to ambiguous issues during compiling. I maintained as it is
Question: when it's recommended to omit or not the type annotation? Is it just to avoid ambiguities or does it help on performance too?

Reply:
It doesn't affect performance, but it makes the function more generic in case we want to support other types of input in the future. Adding type annotation to function arguments without a good reason is a Julia anti-pattern. I did that myself in the past and now whenever I encounter an annotation in the codebase I try to clean it up. See the section "Over-constraining argument types" in this blog post: https://www.oxinabox.net/2020/04/19/Julia-Antipatterns.html